### PR TITLE
fix: Squashed search api bugs

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

File: server/services/v1/query_runner.go [hunk 1]
```diff
@@ -612,6 +617,71 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
+					continue
+				}
+				if cf.DataType != schema.StringType {
+					return nil, api.Errorf(api.Code_INVALID_ARGUMENT, "`%s` is not a searchable field. Only string fields can be queried", sf)
+				}
+				found = true
+				break
+			}
+			if !found {
+				return nil, api.Errorf(api.Code_INVALID_ARGUMENT, "`%s` is not a schema field", sf)
```
Comment (himank) [inline server/services/v1/query_runner.go:654]: nit:  `Field '%s' is not present in the collection`

File: server/services/v1/query_runner.go [hunk 2]
```diff
@@ -612,6 +617,71 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
+		}
+	}
+	return searchFields, nil
+}
+
+func (runner *SearchQueryRunner) getFacetFields(collFields []*schema.Field) (qsearch.Facets, error) {
+	var empty = qsearch.Facets{}
+	facets, err := qsearch.UnmarshalFacet(runner.req.Facet)
+	if err != nil {
+		return empty, err
```
Comment (himank) [inline server/services/v1/query_runner.go:665]: return `qsearch.Facets{}, err`

File: server/services/v1/query_runner.go [hunk 3]
```diff
@@ -612,6 +617,71 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
+	}
+
+	for _, ff := range facets.Fields {
+		found := false
+		for _, cf := range collFields {
+			if ff.Name != cf.FieldName {
```
Comment (himank) [inline server/services/v1/query_runner.go:671]: nit: just move `found=true` inside if 
```
for ... {
  if ff.Name == cf.FieldName {
    found = true
    break
  }
}
```

**Review by himank — APPROVED** [review]

File: server/services/v1/query_runner.go [hunk 4]
```diff
@@ -671,11 +670,15 @@ func (runner *SearchQueryRunner) getFacetFields(collFields []*schema.Field) (qse
 			if ff.Name != cf.FieldName {
 				continue
 			}
+			if cf.DataType != schema.StringType {
+				return qsearch.Facets{}, api.Errorf(api.Code_INVALID_ARGUMENT, "Cannot generate facets for `%s`. Faceting is supported for text fields", ff)
```
Comment (adilansari) [inline server/services/v1/query_runner.go:674]: Lets just release the client with this until we figure out stats unmarshalling.

File: server/services/v1/query_runner.go [hunk 5]
```diff
@@ -671,11 +670,15 @@ func (runner *SearchQueryRunner) getFacetFields(collFields []*schema.Field) (qse
 			if ff.Name != cf.FieldName {
 				continue
 			}
+			if cf.DataType == schema.Int64Type || cf.DataType == schema.Int32Type || cf.DataType == schema.DoubleType {
```
Comment (adilansari) [inline server/services/v1/query_runner.go:673]: Lets release client and temporarily disable faceting for these

File: server/services/v1/query_runner.go [hunk 6]
```diff
@@ -612,6 +617,74 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
+		return qsearch.Facets{}, err
+	}
+
+	for _, ff := range facets.Fields {
+		found := false
+		for _, cf := range collFields {
+			if ff.Name != cf.FieldName {
+				continue
+			}
+			if cf.DataType == schema.Int64Type || cf.DataType == schema.Int32Type || cf.DataType == schema.DoubleType {
```
Comment (himank) [inline server/services/v1/query_runner.go:673]: if `cf.DataType != schema.StringType`?

File: server/services/v1/query_runner.go [hunk 7]
```diff
@@ -612,6 +617,74 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 	return &Response{}, ctx, nil
 }
 
+func (runner *SearchQueryRunner) validateRequestParams() error {
+	if runner.req.PageSize < 0 {
```
Comment (himank) [inline server/services/v1/query_runner.go:621]: we have request related check inside validators.go in api package

**Review by himank — APPROVED** [review]

**Pull request merged**